src/IO/ReadBufferFromS3.cpp

```diff
@@ -162,13 +163,21 @@ bool ReadBufferFromS3::nextImpl()
     if (!next_result)
     {
         read_all_range_successfully = true;
+        // release result to free pooled HTTP session for reuse
+        impl->releaseResult();
         return false;
     }
 
     BufferBase::set(impl->buffer().begin(), impl->buffer().size(), impl->offset());
 
     ProfileEvents::increment(ProfileEvents::ReadBufferFromS3Bytes, working_buffer.size());
     offset += working_buffer.size();
+
+    // release result if possible to free pooled HTTP session for better reuse
+    bool isReadUntilPosition = read_until_position && read_until_position == offset;
```

Comment:
```suggestion
    bool is_read_until_position = read_until_position && read_until_position == offset;
```
(by our style guide)